zulip_bots/zulip_bots/create_bot.py

```diff
@@ -0,0 +1,139 @@
+
+This is a boilerplate bot that can be used as a template for more
+sophisticated/evolved Zulip bots that can be installed separately.
+"""
+
+
+README_TEMPLATE = """This is a boilerplate package for a Zulip bot that can be installed from pip
+and launched using the `zulip-run-bots` command.
+"""
+
+SETUP_TEMPLATE = """import {bot_name}
+from setuptools import find_packages, setup
+
+package_info = {{
+    "name": "{bot_name}",
+    "version": {bot_name}.__version__,
+    "entry_points": {{
+        "zulip_bots.registry": ["{bot_name}={bot_name}.{bot_name}"],
+    }},
+    "packages": find_packages(),
+}}
+
+setup(**package_info)
+"""
+
+BOT_MODULE_TEMPLATE = """# See readme.md for instructions on running this code.
+from typing import Any, Dict
+
+import {bot_name}
+
+from zulip_bots.lib import BotHandler
+
+__version__ = {bot_name}.__version__
+
+
+class {handler_name}:
+    def usage(self) -> str:
+        return \"""
+        This is a boilerplate bot that responds to a user query with
+        "beep boop", which is robot for "Hello World".
+
+        This bot can be used as a template for other, more
+        sophisticated, bots that can be installed separately.
+        \"""
+
+    def handle_message(self, message: Dict[str, Any], bot_handler: BotHandler) -> None:
+        content = "beep boop"  # type: str
+        bot_handler.send_reply(message, content)
+
+        emoji_name = "wave"  # type: str
+        bot_handler.react(message, emoji_name)
+
+
+handler_class = {handler_name}
+"""
+
+
+def create_bot_file(path: Path, file_name: str, content: str) -> None:
+    with open(Path(path, file_name), "w") as file:
+        file.write(content)
+
+
+def parse_args() -> argparse.Namespace:
+    usage = """
+        zulip-create-bot <bot_name>
+        zulip-create-bot --help
+        """
+
+    parser = argparse.ArgumentParser(usage=usage, description="Create a minimal Zulip bot package.")
+
+    parser.add_argument("bot", help="the name of the bot to be created")
+
+    parser.add_argument("--output", "-o", help="the target directory for the new bot", default=".")
+
+    parser.add_argument(
+        "--force",
+        "-f",
+        action="store_true",
+        help="forcibly overwrite the existing files in the output directory",
+    )
+
+    parser.add_argument("--quiet", "-q", action="store_true", help="turn off logging output")
+
+    args = parser.parse_args()
+
+    if not args.bot.isidentifier():
+        parser.error(f'"{args.bot}" is not a valid Python identifier')
+
+    if args.output is not None and not os.path.isdir(args.output):
+        parser.error(f"{args.output} is not a valid path")
+
+    return parser.parse_args()
+
+
+def main() -> None:
+    args = parse_args()
+
+    handler_name = f'{args.bot.title().replace("_", "")}Handler'
+
+    bot_path = Path(args.output, args.bot)
+    bot_module_path = Path(bot_path, args.bot)
+
+    try:
+        os.mkdir(bot_path)
```

Comment:
Reading this again, is there a reason not to use the pathlib functions instead, if we're already using Path? That is, can we drop `import os`?